File: 3-Regresion polinomica/regression_template.R
# Plantilla de regresion

#import dataset
dataset = read.csv("Position_Salaries.csv")
dataset = dataset[,2:3]

#divide dataset into training and testing
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$Salary, SplitRatio = 2/3)
# training_set = subset(dataset, split==TRUE)
# testing_set = subset(dataset, split==FALSE)

#Scaling values
#training_set[,2:3] = scale(training_set[,2:3])
#testing_set[,2:3] = scale(testing_set[,2:3]) 



#Ajustar el modelo de regresion n
# --code goes here --
# regression = lm(...)

#Predecir resultados con el modelo de regresion lineal
#y_pred_poly = predict(regression, newdata = data.frame(Level = 6.5, 
# Level2 = 6.5^2, 
# Level3 = 6.5^3, 
# Level4 = 6.5^4, 
# Level5 = 6.5^5))

#Visualizar la regresion
library(ggplot2)

x_grid = seq(min(dataset$Level), max(dataset$Level),0.1) # para suavizar la curva

ggplot() +
  geom_point(aes(x=dataset$Level, y=dataset$Salary),
             colour="red") +
  geom_line(aes(x=x_grid,
                y=predict(regression, newdata = data.frame(Level = x_grid,
                                                         Level2 = x_grid^2, 
                                                         Level3 = x_grid^3, 
                                                         Level4 = x_grid^4, 
                                                         Level5 = x_grid^5))),
            colour="blue") +
  ggtitle("Modelo de regresión ...")+
  xlab("...")+
  ylab("... ($)")

File: 3-Regresion polinomica/regression_template.py
# -*- coding: utf-8 -*-
"""
Created on Mon Mar 22 16:14:21 2021

@author: axvargas
"""

# Regresion

#lybraries import
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt

#data import
dataset = pd.read_csv('Position_Salaries.csv')

#Matrix de caracteristicas
X = dataset.iloc[:,1:2].values # Debido a que tiene que ser una matriz y no un vector OJO [:,1] vector. [,1:2] matriz: Pero contienen los mismos datos
y = dataset.iloc[:,-1].values


#divide dataset in training and testing
"""from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

#scaling variables
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test) #El test tiene que ser escalado con la misma tranformacion que el de entranemaiento por eso se usa solo transform
#y no es necesario en este caso porque es una variable de si o no 1 o 0
"""

#Ajustar la regresion con el dataset / Crear modelo de regresion
## --code goes here--
##regression = 


#PRedicciones de nuestro modelos puesto para region manager-Partner: 6-7
## --code goes here--
## ex. regression.predict([[6.5]])
## ex. regression_2.predict(poly_reg.fit_transform([[6.5]]))



#Visualizacion de los resultados Modelo polinomico
#Para suavizar la curva:
X_grid = np.arange(min(X), max(X)+0.1, 0.1)
X_grid  = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, y, color="red")
## plt.plot(X, regression.predict(X), color="blue")
## plt.plot(X_grid, regression_2.predict(poly_reg.fit_transform(X_grid)), color="blue") #OJO que se usa X_poly para realizar la preiccion na mas
plt.title("Modelo de regresión polinómica")
plt.xlabel("Posición del empleado")
plt.ylabel("Sueldo ($)")
plt.show()



